app/scripts/modules/core/src/search/global/GlobalSearch.tsx

```diff
@@ -0,0 +1,409 @@
+        showSearchResults: !showMinLengthWarning,
+        showRecentItems: false,
+        showMinLengthWarning
+      });
+    } else {
+      this.showRecentItems();
+    }
+  }
+
+  private hideResults() {
+    this.setState({
+      showSearchResults: false,
+      showRecentItems: false,
+      showMinLengthWarning: false
+    });
+  }
+
+  private reset() {
+    this.setState({
+      showSearchResults: false,
+      showRecentItems: false,
+      showMinLengthWarning: false,
+      query: '',
+      querying: false,
+      categories: []
+    });
+  }
+
+  private showRecentItems() {
+    this.setState({ showRecentItems: true });
+  }
+
+  private clearFilters(result: ISearchResult) {
+    ReactInjector.clusterFilterService.overrideFiltersForUrl(result);
+  }
+
+  public render() {
+    const {
+      showSearchResults,
+      showRecentItems,
+      showMinLengthWarning,
+      query,
+      querying,
+      categories
+    } = this.state;
+
+    const { searchVersion } = SETTINGS;
+
+    return (
+      <li
+        ref={(ref) => this.container = ref}
+        className="global-search open"
+      >
+        <form className="right global-search">
+          <div className="form-group has-feedback">
+            <div className="input-group">
+              <input
+                ref={(ref) => this.searchField = ref}
+                type="search"
+                className="form-control flat input-sm no-border"
+                placeholder="Search"
+                value={query}
+                onBlur={this.searchFieldBlurred}
+                onFocus={this.showResults}
+                onChange={this.queryChanged}
+                onKeyUp={this.searchFieldKeyUp}
+              />
+              <Tooltip
+                placement="right"
+                template={
+                  <span>Keyboard shortcut: <span className="keyboard-key">/</span></span>
+                }
+              >
+                <i className="glyphicon glyphicon-search form-control-feedback" />
+              </Tooltip>
+            </div>
+          </div>
+        </form>
+        {!!query && !querying && showMinLengthWarning &&
+          <ul className="dropdown-menu" role="menu">
+            <li className="horizontal middle center spinner-section">
+              <span className="error-message">Please enter at least 3 characters</span>
+            </li>
+          </ul>
+        }
+        {!!query && querying &&
+          <ul className="dropdown-menu" role="menu">
+            <li className="horizontal middle center spinner-section">
+              <Spinner size="small"/>
+            </li>
+          </ul>
+        }
+        {!query && showRecentItems &&
+          <RecentlyViewedItems
+            limit={5}
+            Component={({ results: recentCategories }) => {
+              if (!recentCategories.length) {
+                return null;
+              }
+
+              this.resultRefs = range(recentCategories.length).map(() => []);
```

Comment:
nitpick: I think these `range` calls are unnecessary...
```
this.resultRefs = recentCategories.map(() => []);
```

Reply:
👍 wow Friday Erik was really not thinking straight on this one, good thing Tuesday Erik is here to fix his mistakes.